libqtile/widget/check_updates.py

```diff
@@ -104,4 +105,19 @@ def button_press(self, x, y, button):
         # type: (int, int, int) -> None
         base.ThreadedPollText.button_press(self, x, y, button)
         if button == 1 and self.execute is not None:
-            Popen(self.execute, shell=True)
+            self.do_execute()
+
+    def do_execute(self):
+        self._process = Popen(self.execute, shell=True)
+
+        if self.refresh_after_execute:
+            self.timeout_add(1, self.check_updates)
+
+    def check_updates(self):
```

Comment:
Just realised that this method is not brilliantly named. Doesn't clash but is too close to existing code.